intake/source/base.py

```diff
@@ -53,7 +52,133 @@ class NoEntry(AttributeError):
+        return self.entry.gui
+
+    @property
+    def plot(self):
+        """
+        Returns a hvPlot object to provide a high-level plotting API.
+
+        To display in a notebook, be sure to run ``intake.output_notebook()``
+        first.
+        """
+        try:
+            from hvplot import hvPlot
+        except ImportError:
+            raise ImportError("The intake plotting API requires hvplot."
+                              "hvplot may be installed with:\n\n"
+                              "`conda install -c pyviz hvplot` or "
+                              "`pip install hvplot`.")
+        metadata = self.metadata.get('plot', {})
+        fields = self.metadata.get('fields', {})
+        for attrs in fields.values():
+            if 'range' in attrs:
+                attrs['range'] = tuple(attrs['range'])
+        metadata['fields'] = fields
+        plots = self.metadata.get('plots', {})
+        return hvPlot(self, custom_plots=plots, **metadata)
+
+    @property
+    def hvplot(self):
+        """
+        Returns a hvPlot object to provide a high-level plotting API.
+        """
+        return self.plot
+
+
+class PersistMixin:
+    def get_persisted(self):
+        from ..container.persist import store
+        from dask.base import tokenize
+        return store[tokenize(self)]()
+
+    @staticmethod
+    def _persist(source, path, **kwargs):
+        """To be implemented by 'container' sources for locally persisting"""
+        raise NotImplementedError
+
+    @property
+    def has_been_persisted(self):
+        from ..container.persist import store
+        from dask.base import tokenize
+        return tokenize(self) in store
+
+    @property
+    def is_persisted(self):
+        from ..container.persist import store
+        return self.metadata.get('original_tok', None) in store
+
+    def persist(self, ttl=None, **kwargs):
+        """Save data from this source to local persistent storage
+
+        Parameters
+        ----------
+        ttl: numeric, optional
+            Time to live in seconds. If provided, the original source will
+            be accessed and a new persisted version written transparently
+            when more than ``ttl`` seconds have passed since the old persisted
+            version was written.
+        kargs: passed to the _persist method on the base container.
+        """
+        from ..container.persist import PersistStore
+        from dask.base import tokenize
+        if 'original_tok' in self.metadata:
+            raise ValueError('Cannot persist a source taken from the persist '
+                             'store')
+        if ttl is not None and not isinstance(ttl, (int, float)):
+            raise ValueError('Cannot persist using a time to live that is '
+                             f'non-numeric. User-provided ttl was {ttl}')
+        store = PersistStore()
+        out = self._export(store.getdir(self), **kwargs)
```

Comment:
Should `_export` be made public?

Reply:
There is a public `export()` function; this is the structure to indicate which might be overridden.